src/plugin.validation.dns.azure/Azure.cs

```diff
@@ -16,17 +18,33 @@ internal class Azure : DnsValidation<Azure>
     {
         private DnsManagementClient _azureDnsClient;
         private readonly DomainParseService _domainParser;
+        private readonly Uri _resourceManagerEndpoint;
 
         private readonly AzureOptions _options;
         public Azure(AzureOptions options,
             DomainParseService domainParser,
-            LookupClientProvider dnsClient, 
-            ILogService log, 
+            LookupClientProvider dnsClient,
+            ILogService log,
             ISettingsService settings)
             : base(dnsClient, log, settings)
         {
             _options = options;
             _domainParser = domainParser;
+
+            if (!AzureEnvironments.ResourceManagerUrls.TryGetValue(options.AzureEnvironment, out string endpoint))
```

Comment:
This will still crash existing renewals, where `options.AzureEnvironment` is `null`.